cmd/edit/cluster/cmd.go

```diff
@@ -500,6 +518,47 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	/*******  AdditionalAllowedPrincipals *******/
+	updateAdditionalAllowedPrincipals := false
+	if additionalAllowedPrincipals != nil {
+		updateAdditionalAllowedPrincipals = true
+	}
+	if !updateAdditionalAllowedPrincipals && additionalAllowedPrincipals == nil &&
+		interactive.Enabled() {
+		updateAdditionalAllowedPrincipalsValue, err := interactive.GetBool(interactive.Input{
+			Question: "Update additional allowed principals",
+			Default:  updateAdditionalAllowedPrincipals,
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid update-additional-allowed-principals value: %s", err)
+			os.Exit(1)
+		}
+		updateAdditionalAllowedPrincipals = updateAdditionalAllowedPrincipalsValue
+	}
+	if updateAdditionalAllowedPrincipals && interactive.Enabled() {
+		aapInputs, err := interactive.GetString(interactive.Input{
+			Question: "Additional Allowed Principal ARNs",
+			Help:     cmd.Flags().Lookup("additional-allowed-principals").Usage,
+			Default:  cluster.AWS().AdditionalAllowedPrincipals(),
```

Comment:
Same as above, validators.

Reply:
Same reply as my above the edit cluster noProxy code also have similar [if slice is populated validation checks](https://github.com/openshift/rosa/blob/v1.2.40/cmd/edit/cluster/cmd.go#L426-L443).

Reply:
Sounds good, @mikeshng I don't have permission to resolve the thread though